src/Reducer.js

```diff
@@ -48,6 +48,11 @@ function inject(state, action, props, scenes) {
             case POP_ACTION:
                 assert(!state.tabs, "pop() operation cannot be run on tab bar (tabs=true)")
                 return {...state, index:state.index-1, children:state.children.slice(0, -1) };
+            case POP_TO_TOP:
+                assert(state.tabs, "Parent="+state.key+" is not tab bar, popToTop action is not valid");
+                let child = state.children[state.index];
+                let newChild = {...child, ...{children: [child.children[0]], index:0}};
+                return {...state, ...{children:[...state.children.slice(0,state.index), newChild, ...state.children.slice(state.index+1)]}};
```

Comment:
It works dynamically if I change the reducer like this:
```js
const reducerCreate = params=> {
  const defaultReducer = Reducer(params);
  return (state, action)=>{
    if (state) {
      let currentTabIndex = state.index;
      if (state.children[currentTabIndex].name === action.key && action.type === 'jump') {
        return defaultReducer(state, {key: state.children[currentTabIndex][0].name, type:'reset'})
      }
    }
    return defaultReducer(state, action);
  }
};
```

I guess thats less code than my proposal :)